### DISCUSSION TITLE
VU meter

### DISCUSSION THREAD

**Discussion by mvacoss** [tttapa/Control-Surface#1080] [Q&A]
For quite a while I was struggling with this thing but couldn't find any solution to my problem. I want to control 12 leds using the VULEDs example except I want to read the level from master. It says that it is only possible to choose between 1-8 tracks but for my custom midi controller I want the master level only. Is there any possible solution? Anyway I'm absolute newbie to this topic (not even speaking about coding). Basically what I want is to use the VULEDs example except I want to read from the master track.

Any suggestion or tip would be appreciated. Thanks

**Comment by tttapa** [comment]
~~See https://tttapa.github.io/Control-Surface-doc/Doxygen/da/dc1/group__MCU__PB.html. You can use the `MCU::MASTER_VOLUME` constant.~~

**Comment by mvacoss** [comment]
Okay so, like I said I don't know how to program that much so could you please help me with this or write it for me? The track I want to use is master. For now it works like it should however for track 1. Thank you
```
#include <Control_Surface.h>

USBMIDI_Interface midi;

const pin_t dataPin  = 2;  // Pin connected to DS of 74HC595 (14)
const pin_t clockPin = 4;  // Pin connected to SH_CP of 74HC595 (11)
const pin_t latchPin = 3;  // Pin connected to ST_CP of 74HC595 (12)

// Instantiate two daisy-chained shift register with the SPI slave select pin as
// latch pin, most significant bit first, and a total of 16 outputs.
ShiftRegisterOut<16> sreg(dataPin, clockPin, latchPin, MSBFIRST);

uint8_t track = 1;

// Instantiate a VULEDs object with 12 LEDs.
MCU::VULEDs<12> vu {
  sreg.pins().slice<0, 11>(), // first 12 pins of the shift registers
  track,                      // track number for Master (track 9)
  50,                         // how long does it take for the meter to decay
};

// Push button to switch between dot and bar mode.
Button dotBarModeButton = A0;

void setup() {
  Control_Surface.begin();
  dotBarModeButton.begin();
}

void loop() {
  Control_Surface.loop();
  if (dotBarModeButton.update() == Button::Falling) // when button is pressed
    vu.toggleMode(); // toggle between dot and bar mode
}
```

**Reply by tttapa** [reply]
My bad, you cannot use `MASTER_VOLUME` for the VU meters.

As far as I know, what you're after is not possible, because the Mackie Control Universal does not have a VU meter for the master bus. IIRC, there is a mode where the DAW could write the level meters to the MCU's LCD display, but this is something your DAW needs to support, and it's not something I've ever tried.

**Comment by mvacoss** [comment]
Oh okay. Is there some other way to do this exact thing? Like mapping different controller instead of MCU or read the data of the master volume level and then output it for the vu meter?

**Reply by tttapa** [reply]
On the Arduino side, anything is possible. The problem is that it is up to your DAW to actually send the level meter information to the control surface. You'll have to experiment with reverse engineering the DAW output, or even write a custom MIDI controller script or config file for your DAW.

**Comment by mvacoss** [comment]
Thank you. Anyway I'm using FL Studio 21 so I think it should be somehow possible to do so, but for now I don't really know how.

**Reply by tttapa** [reply]
See https://www.image-line.com/fl-studio-learning/fl-studio-online-manual/html/midi_scripting.htm.
